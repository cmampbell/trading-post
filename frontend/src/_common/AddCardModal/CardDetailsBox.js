import React, { useState } from "react";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import CardForm from "../CardForm/CardForm";

/* Returns MUI box component with several input boxes and a button.
*  
*  Rendered in AddCardModal after user has selected a card using <Searchbar/>
*
*  state:
*      card - the active card selected from printings.
*             Initially the first card object element in printings.
*
*  props:
*        printings - printings of card selected with <Searchbar />, returned from API
*        addCard - function used to add card to relevant memory.
*                State for trade page.
*                db for <CardBinder/> pages.
*        handleClose - function to close modal on form submit
*        fields - fields for <CardForm/>
*
*  Displays the name and artwork from the selected card, so user can see visual
*  confirmation that card art from db matches their physical card.
*
*  The artwork will have a black border for non-foil cards, and a rainbow border
*  for foil cards. User can change foil type in <CardForm/>
*
*  Renders <CardForm/>, which requires fields as a prop.
*
*  <CardDetailsBox /> should only care about which card is selected.
*/
const CardDetailsBox = ({ printings, addCard, handleClose, fields }) => {

    const [card, setCard] = useState(printings[0]);

    const borderStyles = card.foil === 'Yes' || card.foil === 'Etched' ?
        {
            border: '5px solid transparent',
            borderImage: 'linear-gradient(to bottom right, #b827fc 0%, #2c90fc 25%, #b8fd33 50%, #fec837 75%, #fd1892 100%)',
            borderImageSlice: 1,
        }
        : {
            border: 5,
            borderColor: 'black',
            borderRadius: '10px'
        };

    return (
            <Box sx=
                {{
                    display: 'flex',
                    flexDirection: 'column',
                    alignItems: 'center',
                    justifyContent: 'flex-start',
                    marginTop: 4,
                }}
            >
                {/* container for image */}
                <Box sx={{
                    height: 250,
                    width: 'auto',
                    overflow: 'hidden',
                    ...borderStyles
                }}>
                    <img
                        src={card.art_uri}
                        alt={`${card.name} art by ${card.artist}`}
                        style={{
                            width: 'auto',
                            height: '100%',
                         }}
                    />
                </Box>
                <Typography variant='subtitle2' sx={{ m: 1, fontSize: '11px' }}>
                    Illustrated by {card.artist}. &#8482; & &copy; Wizards Of The Coast, Inc.
                </Typography>

                <Typography variant='h6'>{card.name}</Typography>
                {/* TO-DO: Add context provider here */}
                <CardForm
                    card={card}
                    setCard={setCard}
                    printings={printings}
                    handleClose={handleClose}
                    fields={fields}
                    addCard={addCard}
                    type={'Add'}
                />
            </Box>
    )
}

export default CardDetailsBox;