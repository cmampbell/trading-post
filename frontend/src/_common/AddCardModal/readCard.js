import { createWorker } from 'tesseract.js';

/* parseSymbolResults takes an array of symbol data from Tesseract results
*  and returns a string based on the words associated with that symbol.
*  [{word: {text: 'Example'}, confidence: 98}, 
*   {word: {text: 'Input'}, confidence: 98},
*   {word: {text: 'Question'}, confidence: 50},]
*     ==> 'Example Input 
*/
const parseSymbolResults = symbols => {
    // If there are no symbols, return message
    if(symbols.length < 1) return 'Unable to read text from photo';

    // wordSet is a container for unique words found in the text,
    // kept in insertion order.
    const wordSet = new Set();

    // We want to track when we find the start of a confident series
    // of symbols, and when we find the end
    let foundStart = false;
    let foundEnd = false;

    // loop ends when we found the end of a series of confident symbols
    // or when we loop through all symbols
    for (let i = 0; !foundEnd && i < symbols.length; i++) {
        let currentSymbol = symbols[i]
        // if tesseract is 96% confident in the symbol result
        if (currentSymbol.confidence >= 96) {
            // we check if the set has the word already
            if (!wordSet.has(currentSymbol.word.text.trim())) {
                // add the word tesseract linked to current symbol to wordSet
                // and trim off any whitespace tesseract may have misread
                wordSet.add(currentSymbol.word.text.trim());
            }
            // set found start
            if (!foundStart) foundStart = true;
        } else if (foundStart && currentSymbol.confidence < 96) {
            foundEnd = true;
        }
    }

    const fullString = [];

    // loop through set in insertion order, push found words
    wordSet.forEach((value)=> fullString.push(value));

    return fullString.join(' ');
}

/* We are using tesseract.js for Optical Character Recognition.
*  Check docs for explanation of createWorker/worker
*  Docs: https://github.com/naptha/tesseract.js#tesseractjs
*/
const readCard = async (image) => {
    const worker = await createWorker({
        // langPath is for traineddata, could train it on magic font
    })
    await worker.loadLanguage('eng');
    await worker.initialize('eng');
    await worker.setParameters({
        // Only return the characters in this string, including spaces, hypens, and apostrophes
        tessedit_char_whitelist: `abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-' `
    });
    let result = await worker.recognize(image);
    await worker.terminate();
    return parseSymbolResults(result.data.symbols);
}

export default readCard;