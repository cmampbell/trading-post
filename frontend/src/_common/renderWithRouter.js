import React, { isValidElement } from "react";
import { render } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router-dom";


// borrowed from: https://webup.org/blog/how-to-avoid-mocking-in-react-router-v6-tests/

export default renderWithRouter = (children, routes=[]) => {
    // options creating routes for valid children
    // else keep children as they are
    // first argument is a full route

    const options = isValidElement(children)
            ? { element: children, path: '/'}
            : children;

    // spread out the options here
    // the options are a route object
    // then routes are spread and passed after
    const router = createMemoryRouter([{...options}, ...routes],
        {
        initialEntries: [options.path],
        initialIndex: 1
    }
    );

    return render(<RouterProvider router={router}/>);
}