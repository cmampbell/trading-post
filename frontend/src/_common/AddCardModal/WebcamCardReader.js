import React, { useRef, useState, useCallback, useEffect } from 'react';
import Webcam from "react-webcam";
import { Image } from 'image-js';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import readCard from './readCard';

/* We are using react-webcam for webcam access. Docs: https://github.com/mozmorris/react-webcam
*  We are using image-js for image pre-processing. Docs: https://image-js.github.io/image-js/#image
*
*  The user can take a photo of their card by finding it with the camera and clicking the
*  capture photo button. Once the photo has been saved in state as imgSrc,
*  our useEffect callback runs.
*
*  We use Tesseract.recognize to scan the card for any english words it can find. For now,
*  we assume that the user has the card facing upright and the title of the card will be the
*  first thing read, because Tesseract will read left-> right, top -> bottom.
*
*  With this assumption we take the resulting array of found words, and find the first word
*  with a confidence score over 89, and the last word with a confidence score over 89,
*  and pass the string of results into getCardWithCamera(), which sends the card name guess
*  back to the searchBar component.
*
*/

const preProcessImage = async (imageSource) => {

    const image = await Image.load(imageSource);

    // might want to resize to give tesseract more to read
    const cropped = image.crop({ x: 80, y: 73, width: 460, height: 146 });
    // const resized = cropped.resize({ width: 820, height: 272 });
    const resized = cropped.resize({ width: 1380, height: 438 });
    const multiplied = resized.multiply(1.7);
    let grey = multiplied.grey();
    let blur = grey.gaussianFilter({ radius: 1 });
    let mask = blur.mask({ threshold: 0.48 });

    // return cropped.toDataURL();
    return mask.toDataURL();
}

const WebcamCardReader = ({ getCardWithCamera, closeCameraModal, setSearchInput }) => {
    const webcamRef = useRef(null);
    const [imgSrc, setImgSrc] = useState(null);
    // add peice of state that tracks if webcam is loading or not
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        if (webcamRef.current) {
            setIsLoading(() => false);
        }
    }, [webcamRef])

    const capture = useCallback(async () => {
        const photo = webcamRef.current.getScreenshot({ height: 1280, width: 720 });
        const processedPhoto = await preProcessImage(photo);
        setImgSrc(()=> processedPhoto);
        const result = await readCard(processedPhoto);
        console.log(result);
        getCardWithCamera(result);
        closeCameraModal();
    }, [webcamRef, setImgSrc]);

    return (
        // the parent box gives the camera it's dimensions
        <Box
            sx={{
                position: 'relative',
                display: 'flex',
                flexDirection: 'column',
                height: '640px',
                width: '360px',
                backgroundColor: 'white',
            }}>
            {isLoading && <Typography variant='subtitle1'> Loading camera...</Typography>}
            {/* This box provides card placement guide */}
            <Box
                sx={{
                    border: 'solid',
                    borderColor: '#d86409',
                    position: 'absolute',
                    height: '440px',
                    width: '304px',
                    top: '50px',
                    left: '30px',
                    zIndex: 2
                }}>
            </Box>
            {/* This box shows the cropped area of the photo, used for testing */}
            {/* <Box sx={{
                            border: 'solid',
                            borderColor: 'red',
                            position: 'absolute',
                            height: '30px',
                            width: '230px',
                            top: '73px',
                            left: '40px',
                            zIndex: 2
                        }}></Box> */}
            <Typography
                variant='h5'
                style={{ backgroundColor: 'white', textAlign: 'center' }}
            >
                Place card borders inside outline
            </Typography>
            <Webcam
                audio={false}
                ref={webcamRef}
                screenshotFormat='image/png'
                videoConstraints={{
                    height: 1920,
                    width: 1080,
                }}
            />
            {!imgSrc && <>
                <Grid container spacing={8} sx={{ position: 'absolute', bottom: '0%' }}>
                    <Grid item xs={6}>
                        <Button
                            onClick={capture}
                            color='primary'
                            variant='contained'
                            sx={{
                                width: '100%'
                            }}
                        >
                            Take Photo
                        </Button>
                    </Grid>
                    <Grid item xs={6}>
                        <Button
                            onClick={closeCameraModal}
                            color='error'
                            variant='contained'
                            sx={{
                                width: '100%'
                            }}
                        >
                            Close Camera
                        </Button>
                    </Grid>
                </Grid>
            </>}

            {/* <img src={imgSrc} alt='card-scan-result' /> */}
        </Box >
    )
}



export default WebcamCardReader;