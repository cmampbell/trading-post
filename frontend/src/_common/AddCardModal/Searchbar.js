import React, { useEffect, useState } from "react";
import TextField from "@mui/material/TextField";
import Autocomplete from "@mui/material/Autocomplete";
import Button from "@mui/material/Button";
import Modal from "@mui/material/Modal";
import Box from "@mui/material/Box";
import IconButton from "@mui/material/IconButton";
import Grid  from "@mui/material/Grid";
import PhotoCameraIcon from '@mui/icons-material/PhotoCamera';
import CardService from "../../Api/CardService"
import WebcamCardReader from "./WebcamCardReader";

/* Returns MUI autocomplete component https://mui.com/material-ui/react-autocomplete/
*
* User begins typing, after half a second after last input change, API call is made
*   matching card names autofill a dropdown menu, user can pick
*   selected card state from AddCardModal is set based on user selection
*       Searchbar cleared
*       selected card is populated into CardDetailsBox on AddCardModal
*
* User can also choose to scan card with their camera. The user captures a photo of
* the card, and the Webcam component calls getCardWithCamera, with the tesseract
* result of what it thinks the card name is. We then set the selectedCard for the
* AddCardModal equal to the first entry of the API results.
*
*   The first entry will be the correct card if tesseract was able to extract it,
*   or the first returned card from the name like search.
*
*/
const Searchbar = ({ setSelectedCard, selectedCard }) => {
    const [searchInput, setSearchInput] = useState('');
    const [cardOptions, setCardOptions] = useState([])
    const [isLoading, setIsLoading] = useState(false);
    const [cameraOpen, setCameraOpen] = useState(false)

    useEffect(() => {
        let timerID;
        if (searchInput && !selectedCard) {
            setIsLoading(true)
            timerID = setTimeout(async () => {
                try {
                    const cards = await CardService.getCardsByName(searchInput)
                    console.log(cards);
                    setCardOptions(() => [...cards])
                    setIsLoading(() => false)
                } catch (err) {
                    console.log(err)
                }
            }, 500);
        }
        return () => {
            // clear ID if search input changes bc user is typing
            clearTimeout(timerID)
            // Set cards to empty array to clear old results
            setCardOptions(() => [])
        }
    }, [searchInput, selectedCard])

    const handleInputChange = (evt, newValue) => {
        // controlled autocomplete
        setSearchInput(() => newValue)
    }

    const handleValueChange = (evt, newValue) => {
        // value changes when user makes a selection
        if (newValue) setSelectedCard(() => newValue)
    }

    const openCameraModal = () => {
        setCameraOpen(() => true)
    }

    const closeCameraModal = () => {
        setCameraOpen(() => false)
    }

    const getCardWithCamera = async (cardNameGuess) => {
        if (cardNameGuess) setSearchInput(() => cardNameGuess);
        try {
            const card = await CardService.getCardsByName(cardNameGuess);
            setSelectedCard(() => ({ name: card[0].name, id: card[0].oracle_id }))
        } catch (err) {
            console.error(err)
        }
    }

    return (
        <>
            <Grid container sx={{width: '100%'}}>
                <Grid item xs={10}>
                    <Autocomplete
                        inputValue={searchInput}
                        onInputChange={handleInputChange}
                        value={selectedCard}
                        onChange={handleValueChange}
                        id="searchbar"
                        freeSolo
                        selectOnFocus
                        handleHomeEndKeys
                        clearOnEscape
                        autoSelect
                        loading={isLoading}
                        loadingText='Loading...'
                        noOptionsText='No cards found.'
                        options={cardOptions.map((card) => ({ id: card.oracle_id, label: card.name }))}
                        renderInput={(params) => <TextField
                            {...params}
                            label="Card Name"
                        />}
                        sx={{ marginLeft: '5%', width: '90%' }}
                        filterOptions={(x) => x}
                    />
                </Grid>
                <Grid item xs={1}>
                    <IconButton onClick={openCameraModal} color="primary" >
                        <PhotoCameraIcon sx={{ fontSize: '35px' }}/>
                    </IconButton>
                </Grid>
            </Grid>

            <Modal
                open={cameraOpen}
                onClose={closeCameraModal}
                aria-labelledby="camera-modal-title"
                aria-describedby="camera-modal"
            >
                <Box
                    sx={{
                        display: 'flex',
                        justifyContent: 'flex-start',
                        flexDirection: 'column',
                        alignItems: 'center',
                        marginTop: '8vh',
                        height: '85vh',
                        position: 'relative'

                    }}>
                    <WebcamCardReader
                        setSearchInput={setSearchInput}
                        closeCameraModal={closeCameraModal}
                        getCardWithCamera={getCardWithCamera}
                    />
                    <Button></Button>
                </Box>
            </Modal>
        </>
    )
}

export default Searchbar;