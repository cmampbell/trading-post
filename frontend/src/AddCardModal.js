import React, { useState, useEffect } from 'react';
import { Modal, Container, Paper, Button, IconButton, Box } from "@mui/material";
import CloseIcon from '@mui/icons-material/Close';
import Searchbar from './Searchbar'
import CardDetailsBox from './CardDetailsBox';
import Api from './Api';

/* Returns MUI modal component https://mui.com/material-ui/react-modal/
*
*  Contains Searchbar component and CardDetailsBox component
*  
*  This tracks the selected card returned from Searchbar
*
*  Once the user makes a selection with Searchbar,
*  this makes a call to API to get all printings of card
*
*  Once we have the printings, we display the CardDetailsBox using
*  the card objects returned from the API to autofill the inputs
*
*  On close, we clear everything from state, allowing the user to
*  start a new search. Tradeoff is that user will lose all data
*  if they click out of the modal accidentally
*
*/

const AddCardModal = ({ open, setListCards, setSearchOpen }) => {

    const [selectedCard, setSelectedCard] = useState('')
    const [printings, setPrintings] = useState([])

    useEffect(() => {
        if (selectedCard && selectedCard.id) {
            // get matching card printings from database
            Api.getCardsByOracleId(selectedCard.id)
                .then((options) => {
                    // set selected options to returned cards
                    setPrintings(() => [...options])
                })
        }
        return () => {
            setSelectedCard('')
            setPrintings([])
        }
    }, [selectedCard])

    const handleClose = () => {
        setPrintings([]);
        setSearchOpen(false)
    }

    return (
        <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="modal-title"
            aria-describedby="modal-description"
        >
            <Container maxWidth="md">
                <Paper elevation={8}
                    sx={{
                        display: 'flex',
                        justifyContent: 'flex-start',
                        flexDirection: 'column',
                        alignItems: 'center',
                        marginTop: '8vh',
                        height: '85vh'
                    }}
                >
                    <Box
                        sx={{
                            display: 'flex',
                            justifyContent: 'center',
                            flexDirection: 'row',
                            width: '100%',
                            position: 'relative'
                        }}>
                        <h2 id='modal-title'>Search By Card Name</h2>
                        <IconButton
                            aria-label="delete"
                            onClick={handleClose}
                            sx={{
                                position: 'absolute',
                                top: '25%',
                                right: '3.5%'
                            }}
                        >
                            <CloseIcon />
                        </IconButton>
                    </Box>
                    <Searchbar
                        setSelectedCard={setSelectedCard}
                        selectedCard={selectedCard}
                    />
                    {printings.length > 0
                        && <CardDetailsBox
                            cards={printings}
                            setListCards={setListCards}
                            handleClose={handleClose}
                        />}
                </Paper>
            </Container>
        </Modal >
    )
}

export default AddCardModal;