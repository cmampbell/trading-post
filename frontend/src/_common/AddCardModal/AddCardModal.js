import React, { useState, useEffect } from 'react';
import { Modal, Container, Paper, IconButton, Box, Typography } from "@mui/material";
import CloseIcon from '@mui/icons-material/Close';
import Searchbar from './Searchbar'
import CardDetailsBox from './CardDetailsBox';
import CardService from '../../Api/CardService';

/* Returns MUI modal component https://mui.com/material-ui/react-modal/
*
*  Contains Searchbar component and CardDetailsBox component
*  
*  This tracks the selected card returned from Searchbar
*
*  Once the user makes a selection with Searchbar,
*  this makes a call to API to get all printings of card
*
*  Once we have the printings, we display the CardDetailsBox using
*  the card objects returned from the API to autofill the inputs
*
*  On close, we clear everything from state, allowing the user to
*  start a new search. Tradeoff is that user will lose all data
*  if they click out of the modal accidentally
*
*/

const AddCardModal = ({ open, addCard, setSearchOpen, fields }) => {

    const [selectedCard, setSelectedCard] = useState()
    const [printings, setPrintings] = useState([])

    useEffect(() => {
        if (selectedCard && selectedCard.id) {
            // get matching card printings from database
            CardService.getCardsByOracleId(selectedCard.id)
                .then((options) => {
                    // set selected options to returned cards
                    setPrintings(() => [...options])
                })
        }
        return () => {
            setSelectedCard('')
            setPrintings([])
        }
    }, [selectedCard])

    const handleClose = () => {
        setPrintings([]);
        setSearchOpen(false)
    }

    return (
        <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="modal-title"
            aria-describedby="modal-description"
        >
            <Container maxWidth="md">
                <Paper elevation={8}
                    sx={{
                        display: 'flex',
                        justifyContent: 'flex-start',
                        flexDirection: 'column',
                        alignItems: 'center',
                        marginTop: '6vh',
                        height: '88vh',
                        position: 'relative',
                        width: '100%'
                    }}
                >
                    <Box
                        sx={{
                            display: 'flex',
                            justifyContent: 'center',
                            flexDirection: 'row',
                            width: '100%',
                            position: 'relative'
                        }}>
                        <Typography variant='h2' sx={{m: 2}}>
                            Card Search
                        </Typography>
                        <IconButton
                            aria-label="delete"
                            onClick={handleClose}
                            color='error'
                            sx={{
                                position: 'absolute',
                                top: '25%',
                                right: '3.5%'
                            }}
                        >
                            <CloseIcon />
                        </IconButton>
                    </Box>
                    <Searchbar
                        setSelectedCard={setSelectedCard}
                        selectedCard={selectedCard}
                    />
                    {printings.length > 0
                        && <CardDetailsBox
                            printings={printings}
                            addCard={addCard}
                            handleClose={handleClose}
                            fields={fields}
                        />}
                </Paper>
            </Container>
        </Modal >
    )
}

export default AddCardModal;