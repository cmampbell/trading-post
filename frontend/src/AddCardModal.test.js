import React from "react";
import axios from 'axios'

import renderWithRouter from "./renderWithRouter";
import AddCardModal from "./AddCardModal";

import userEvent from '@testing-library/user-event'
import { act } from "react-dom/test-utils";
import { waitFor, render } from "@testing-library/react";

const testCard = { name: 'Island', image_uri: 'test_uri', usd_price: '7.65', id: '1' , setCode: 'IXL'}

// mock axios for useEffect API call
jest.mock('axios');

describe('AddCardModal tests', () => {
    it('should render without crashing', () => {
        render(<AddCardModal open={true} onClose={()=> false}/>);
    })

    it('should match snapshot', () => {
        const { asFragment } = render(<AddCardModal open={true} onClose={()=> null}/>);

        expect(asFragment()).toMatchSnapshot();
    })

    it('should list returned cards after user types, clear cards after search term is cleared', async () => {
        axios.mockResolvedValue({data:{cards: [testCard]}});
        const { queryByText, queryByLabelText } = renderWithRouter(<AddCardModal open={true} onClose={()=> false}/>);

        const textInput = queryByLabelText('Card Name')
        expect(queryByText('Add Cards')).toBeInTheDocument();
        expect(textInput).toBeInTheDocument();

        await act(async () => {
            await userEvent.type(textInput, 'isl');
        })

        expect(textInput).toHaveValue('isl')

        await waitFor(() => {
            expect(queryByText("Island")).toBeInTheDocument()
        })

        await act(async ()=> {
            await userEvent.clear(textInput)
        })

        await waitFor(() => {
            expect(queryByText("Island")).not.toBeInTheDocument()
        })        
    })

    it('should return no cards if no cards found', async () => {
        axios.mockResolvedValue({data:{cards: []}});
        const { queryByText, queryByLabelText } = renderWithRouter(<AddCardModal open={true} onClose={()=> false}/>);

        const textInput = queryByLabelText('Card Name')
        expect(queryByText('Add Cards')).toBeInTheDocument();
        expect(textInput).toBeInTheDocument();

        await act(async () => {
            await userEvent.type(textInput, 'not-a-card');
        })

        expect(textInput).toHaveValue('not-a-card')

        await waitFor(() => {
            expect(queryByText("Island")).not.toBeInTheDocument()
        })
    })
})