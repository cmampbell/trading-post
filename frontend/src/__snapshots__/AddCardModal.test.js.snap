// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardModal tests should match snapshot 1`] = `<DocumentFragment />`;
